services/release/release.go

```diff
@@ -23,6 +23,23 @@ func createTag(gitRepo *git.Repository, rel *models.Release, msg string) (bool,
 	// Only actual create when publish.
 	if !rel.IsDraft {
 		if !gitRepo.IsTagExist(rel.TagName) {
+			if rel.Repo != nil {
```

Comment:
This feels fragile - I bet there are plenty of codepaths passed in that don't always set the repository.

```go
if rel.RepoID != 0 { 
   if err = rel.LoadRepo(); err != nil {
     log.Error("Unable to load repository[%d] for release[%d] %s. Error: %v", rel.RepoID, rel.ID, rel.TagName, err)
     return false, err
   }
   // then the rest of this stuff
```

Or even simpler just make `GetProtectedTagsByRepoID(...)`

Reply:
I checked the paths and assigned the repo. I just moved the `LoadAttributes` call to the top now.

Reply:
There's no point loading the repository - you're not actually using it all.

Reply:
It is used in line 31 and 58.